common/reconciliation/store/blobstorewriter_test.go

```diff
@@ -0,0 +1,99 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package store
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/uber/cadence/common/blobstore"
+	"github.com/uber/cadence/common/blobstore/filestore"
+	"github.com/uber/cadence/common/config"
+)
+
+func TestBlobstoreWriter(t *testing.T) {
+	uuid := "test-uuid"
+	extension := Extension("test")
+	outputDir := t.TempDir()
+
+	cfg := &config.FileBlobstore{
+		OutputDirectory: outputDir,
+	}
+	//Reusing the Filestoreclient from the other sister test in the same package.
+	blobstoreClient, err := filestore.NewFilestoreClient(cfg)
+	require.NoError(t, err)
+
+	blobstoreWriter := NewBlobstoreWriter(uuid, extension, blobstoreClient, 10).(*blobstoreWriter)
+
+	type testCase struct {
+		name        string
+		input       string
+		expectedErr bool
+	}
+
+	//Trying a table test approach this time.
+	testCases := []testCase{
+		{
+			name:        "Normal case",
+			input:       "test-data",
+			expectedErr: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Add data to the writer
+			err := blobstoreWriter.Add(tc.input)
+			if tc.expectedErr {
+				assert.Error(t, err)
+			} else {
```

Comment:
avoid extra indentation by handling special case and returning.
```
if tc.expectedErr {
  assert.Error(t, err)
  return
}

// rest of the code goes here without an else { } wrapper
```

Reply:
this would be an ideal case for `require.NoError(t, err)` fwiw.  that'll stop the test immediately.

(this is also one of the reasons why embedding Assertions is problematic.  it hides whether things are asserts or requires, and that's a _VERY_ important detail for this reason and for concurrency-safety reasons (you cannot safely `require` on a background thread))